# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/builtin/FinalizationGroupObject.h
# Commit: b1117df3b91d
# Full Hash: b1117df3b91d10d68d10b399637cf9ac295057e6
# Author: Jon Coppeard <jcoppeard@mozilla.com>
# Date: 2019-11-01 16:13:11
# Regressor Bug: 1587096
# File Overlap Count: 1
# Description:
#   Bug 1587096 - Part 3: Implmement FinalizationGroup r=sfink
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D49946
# ==============================================================================

diff -r cefc0474e13c -r b1117df3b91d js/src/builtin/FinalizationGroupObject.h
--- a/js/src/builtin/FinalizationGroupObject.h	Fri Nov 01 10:37:40 2019 +0000
+++ b/js/src/builtin/FinalizationGroupObject.h	Fri Nov 01 10:37:47 2019 +0000
@@ -173,6 +173,17 @@
   static const JSPropertySpec properties_[];
 
   static bool construct(JSContext* cx, unsigned argc, Value* vp);
+  static bool register_(JSContext* cx, unsigned argc, Value* vp);
+  static bool unregister(JSContext* cx, unsigned argc, Value* vp);
+  static bool cleanupSome(JSContext* cx, unsigned argc, Value* vp);
+
+  static bool addRegistration(JSContext* cx,
+                              HandleFinalizationGroupObject group,
+                              HandleObject unregisterToken,
+                              HandleFinalizationRecordObject record);
+  static void removeRegistrationOnError(HandleFinalizationGroupObject group,
+                                        HandleObject unregisterToken,
+                                        HandleFinalizationRecordObject record);
 
   static void trace(JSTracer* trc, JSObject* obj);
   static void finalize(JSFreeOp* fop, JSObject* obj);
@@ -192,6 +203,9 @@
   FinalizationGroupObject* finalizationGroup() const;
   size_t index() const;
 
+  void incIndex();
+  void clearFinalizationGroup();
+
  private:
   friend class GlobalObject;
   static const JSFunctionSpec methods_[];